components/measure-upload/MeasureRepositoryUpload.tsx

```diff
@@ -0,0 +1,235 @@
+import { Button, Collapse, Grid, Group, Loader, Select, TextInput } from '@mantine/core';
+import React, { useState } from 'react';
+import { useEffect } from 'react';
+import { useRecoilState, useRecoilValue, useSetRecoilState } from 'recoil';
+import { measureBundleState } from '../../state/atoms/measureBundle';
+import { measurementPeriodState } from '../../state/atoms/measurementPeriod';
+import { identifyMissingValueSets, MeasureUploadProps, populateMeasurementPeriod } from '../../util/measureUploadUtils';
+import { showNotification } from '@mantine/notifications';
+import { v4 as uuidv4 } from 'uuid';
+import { IconAlertCircle } from '@tabler/icons';
+import { CircleCheck } from 'tabler-icons-react';
+import { displayMapToSelectDataState } from '../../state/selectors/displayMapToSelectData';
+
+enum PACKAGE_STATES {
+  LOADING,
+  SUCCESS,
+  FAIL,
+  NONE
+}
+
+const MEASURE_RETRIEVAL_BODY = {
+  resourceType: 'Parameters',
+  parameter: [
+    {
+      name: 'include-terminology',
+      valueBoolean: true
+    }
+  ]
+};
+
+const MEASURE_RETRIEVAL_OPTIONS = {
+  method: 'POST',
+  body: JSON.stringify(MEASURE_RETRIEVAL_BODY),
+  headers: { 'Content-Type': 'application/json+fhir' }
+};
+
+export default function MeasureRepositoryUpload({ logError }: MeasureUploadProps) {
+  const [isLoadingIds, setIsLoadingIds] = useState(false);
+  const [isLoadingPackage, setIsLoadingPackage] = useState<PACKAGE_STATES>(PACKAGE_STATES.NONE);
+  const [measureSelectIsOpen, setMeasureSelectIsOpen] = useState(false);
+
+  const [{ measureRepositoryUrl, isFile, selectedMeasureId, displayMap }, setMeasureBundle] =
+    useRecoilState(measureBundleState);
+  const setMeasurementPeriod = useSetRecoilState(measurementPeriodState);
+
+  const idSelectData = useRecoilValue(displayMapToSelectDataState);
+
+  useEffect(() => {
+    if (isFile) {
+      setMeasureSelectIsOpen(false);
+      setIsLoadingPackage(PACKAGE_STATES.NONE);
+    }
+  }, [isFile]);
+
+  const rejectUpload = (
```

Comment:
Would it be possible to make this a bit more reusable between `MeasureRespositoryUpload.tsx` and `MeasureFileUpload.tsx` and put it in `measureUploadUtils.ts`?